apps/marks/tests.py

```diff
@@ -64,3 +68,143 @@ def testGettingExpirationDateWhenDateInSummerVacation(self):
+
+    def test_creating_a_ruleset_for_the_past_does_not_replace_current(self):
+        past_date = timezone.now() - timezone.timedelta(days=2)
+        G(MarkRuleSet, valid_from_date=past_date)
+
+        self.assertEqual(MarkRuleSet.get_current_rule_set(), self.rule_set)
+
+    def test_changing_an_old_ruleset_does_not_replace_the_current(self):
+        new_rule_set: MarkRuleSet = G(MarkRuleSet)
+        self.rule_set.version = '1.0.1'
+        self.rule_set.content = 'Changed the old rules'
+        self.rule_set.save()
+
+        self.assertEqual(MarkRuleSet.get_current_rule_set(), new_rule_set)
+
+
+class MarkRuleAcceptanceTest(TestCase):
+
+    def setUp(self):
+        self.rule_set: MarkRuleSet = G(MarkRuleSet, version='1.0.0')
+        self.user: User = G(User)
+
+    def test_accepting_mark_rules_sets_them_as_accepted(self):
+        MarkRuleSet.accept_mark_rules(self.user)
+
+        self.assertTrue(self.user.mark_rules_accepted)
+
+    def test_creating_new_mark_rules_sets_accepted_to_false(self):
+        MarkRuleSet.accept_mark_rules(self.user)
+        self.assertTrue(self.user.mark_rules_accepted)
+
+        G(MarkRuleSet, valid_from_date=timezone.now())
+        self.assertFalse(self.user.mark_rules_accepted)
+
+
+class MarkRuleSetAPITest(OIDCTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.user: User = G(User, username='test_user')
+        self.token = self.generate_access_token(self.user)
+        self.headers = {
+            **self.generate_headers(),
+            **self.bare_headers,
+        }
+
+        self.url = reverse('mark_rule_sets-list')
+        self.id_url = lambda _id: reverse('mark_rule_sets-detail', args=[_id])
+
+        self.rule_set: MarkRuleSet = G(MarkRuleSet, version='1.0.0')
+
+    def test_marks_rule_sets_api_returns_ok_without_auth(self):
+        response = self.client.get(self.url, **self.bare_headers)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_un_authenticated_user_can_view_mark_rules(self):
+        response = self.client.get(self.id_url(self.rule_set.id), **self.bare_headers)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+
+class MarkRuleAcceptanceAPITest(OIDCTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.user: User = G(User, username='test_user')
+        self.token = self.generate_access_token(self.user)
+        self.headers = {
+            **self.generate_headers(),
+            **self.bare_headers,
+        }
+
+        self.url = reverse('mark_rule_acceptance-list')
+        self.id_url = lambda _id: reverse('mark_rule_acceptance-detail', args=[_id])
+
+        self.rule_set: MarkRuleSet = G(MarkRuleSet, version='1.0.0')
+
+    def create_rule_acceptance(self):
+        self.rule_acceptance: RuleAcceptance = G(RuleAcceptance, user=self.user, rule_set=self.rule_set)
+
+    def test_marks_rule_acceptance_api_returns_ok_without_auth(self):
```

Comment:
Isn't this function wrongly named? `returns_ok` => `returns_403` / `returns_forbidden`